docs/apps/lnw/es2k/es2k-linux-networking-lag.md

```diff
@@ -0,0 +1,336 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+- -->
+
+# Linux Networking with LAG (ES2K)
+
+This document explains how to run the Linux networking scenario with LAG(Link Aggregation Group) on ES2K.
+
+## Topology
+
+![Linux Networking Topology](es2k-ecmp-topology.png)
+
+Notes about topology:
+
+- Four Kernel netdevs are created by default by loading IDPF driver during ACC bring-up. You can also create more than Four netdevs. For that, we need to modify `acc_apf` parameter under `num_default_vport` in `/etc/dpcp/cfg/cp_init.cfg` on IMC before starting `run_default_init_app`.
+- In `/etc/dpcp/cfg/cp_init.cfg` file also modify default `sem_num_pages` value to the value mentioned in `/opt/p4/p4sde/share/mev_reference_p4_files/linux_networking/README_P4_CP_NWS`.
+- In `/etc/dpcp/cfg/cp_init.cfg` file also modify default `allow_change_mac_address` value to true.
+- vlan1, vlan2, .... vlanN created using Linux commands and are on top of an IDPF Netdev. These VLAN ports should be equal to number of VM's that are spawned.
+- br-int, VxLAN ports are created using ovs-vsctl command provided by the networking recipe and all the vlan ports are attached to br-int using ovs-vsctl command.
+- Both physical ports P0 and P1 should be B2B connected to the Link Partner device and LAG should be configured on both the devices with the associated ports as LAG members using ip link command.
+
+System under test will have above topology running the networking recipe. Link Partner can have the networking recipe or legacy OvS or kernel VxLAN. Note the [Limitations](#limitations) section before setting up the topology.
+
+## Create P4 artifacts and start Infrap4d process
+
+- Use Linux networking p4 program present in the directory `/opt/p4/p4sde/share/mev_reference_p4_files/linux_networking` for this scenario.
+- See [Running Infrap4d on Intel IPU E2100](/guides/es2k/running-infrap4d) for compiling `P4 artifacts`, `bringing up ACC` and running `infrap4d` on ACC.
+
+## Creating the topology
+
+The p4rt-ctl and ovs-vsctl utilities can be found in $P4CP_INSTALL/bin.
+
+### Set the forwarding pipeline
+
+Once the application is started, set the forwarding pipeline config using
+P4Runtime Client `p4rt-ctl` set-pipe command
+
+```bash
+$P4CP_INSTALL/bin/p4rt-ctl set-pipe br0 $OUTPUT_DIR/linux_networking.pb.bin \
+    $OUTPUT_DIR/linux_networking.p4info.txt
+```
+
+Note: Assuming `linux_networking.pb.bin` and `linux_networking.p4info.txt`
+along with other P4 artifacts are created as per the steps mentioned in previous section.
+
+### Configure VSI Group and add a netdev
+
+Use one of the IDPF netdevs on ACC to receive all control packets from overlay
+VM's by assigning to a VSI group. VSI group 3 is dedicated for this configuration,
```

Comment:
```suggestion
VMs by assigning to a VSI group. VSI group 3 is dedicated for this configuration,
```
(no apostrophe for VMs). Its used as a plural, not a posession.

Reply:
same applies for `VM's` and `VF's` used below in rest of document.

Reply:
Addressed.